codec/encode.go

```diff
@@ -0,0 +1,96 @@
+package codec
+
+import (
+	"encoding/binary"
+	"errors"
+)
+
+// Encode is the top-level function which performs SCALE encoding of b which may be of type []byte, int16, int32, int64,
+// or bool
+func Encode(b interface{}) ([]byte, error) {
+	switch v := b.(type) {
+	case []byte:
+		return encodeByteArray(v)
+	case int16:
+		return encodeInteger(int(v))
+	case int32:
+		return encodeInteger(int(v))
+	case int64:
+		return encodeInteger(int(v))
+	case bool:
+		return encodeBool(v)
+	default:
+		return nil, errors.New("Unsupported type!")
```

Comment:
Errors should not start with capitals nor end with punctuation (https://github.com/golang/go/wiki/Errors)

Reply:
fixed